pkg/scheduler/queue/tenant_queues.go

```diff
@@ -144,13 +150,13 @@ func (qb *queueBroker) enqueueRequestBack(r requestToEnqueue) error {
 //
 // max tenant queue size checks are skipped even though queue size violations
 // are not expected to occur when re-enqueuing a previously dequeued request.
-func (qb *queueBroker) enqueueRequestFront(r requestToEnqueue) error {
-	queue, err := qb.getOrAddTenantQueue(r.tenantID, r.maxQueriers)
+func (qb *queueBroker) enqueueRequestFront(request *tenantRequest) error {
+	queue, err := qb.getOrAddTenantQueue(request.tenantID, request.maxQueriers)
```

Comment:
This wasn't changed as part of this PR, but just noticed this now: should we be passing `maxQueriers` here? I think this would potentially reset the number of queriers to an incorrect value. For example, imagine the following sequence of events:

1. Query request A passed to `enqueueRequestBack`, with max queriers set to 20
2. Query request B passed to `enqueueRequestBack`, with max queriers set to new value of 30
3. Query request A is dequeued, but can't be sent to a querier, so `enqueueRequestFront` is called with query request A and max queriers set to 20

Step 3 above would reset the number of queriers for the tenant back to 20, but we want it to remain as 30, right?

Reply:
this is a good call, will change in upcoming PRs